hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/ValueQueue.java

```diff
@@ -269,12 +269,23 @@ public ValueQueue(final int numValues, final float lowWaterMark, long expiry,
    * Initializes the Value Queues for the provided keys by calling the
    * fill Method with "numInitValues" values
    * @param keyNames Array of key Names
-   * @throws ExecutionException executionException.
+   * @throws IOException if no successful initialization for any key
```

Comment:
The wording here is confusing. One way to read this is if any key. fails to initialize, then an except will be thrown. But IIUC an exception will be thrown if all keys fail to initialize.

Reply:
updated.